core/src/test/java/com/exadel/kaliada/core/services/impl/HarvardNewsSearchServiceImplTest.java

```diff
@@ -0,0 +1,93 @@
+package com.exadel.kaliada.core.services.impl;
+
+import com.day.cq.tagging.InvalidTagFormatException;
+import com.day.cq.tagging.Tag;
+import com.day.cq.tagging.TagManager;
+import com.day.cq.wcm.api.Page;
+import com.exadel.kaliada.core.models.NewsModel;
+import io.wcm.testing.mock.aem.MockTagManager;
+import io.wcm.testing.mock.aem.junit5.AemContext;
+import io.wcm.testing.mock.aem.junit5.AemContextExtension;
+import org.apache.sling.testing.mock.sling.ResourceResolverType;
+import org.apache.sling.testing.mock.sling.oak.OakMockSlingRepository;
+import org.apache.sling.testing.mock.sling.servlet.MockRequestPathInfo;
+import org.apache.sling.testing.mock.sling.servlet.MockSlingHttpServletRequest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith(AemContextExtension.class)
+class HarvardNewsSearchServiceImplTest {
+
+    private final AemContext aemContext = new AemContext(ResourceResolverType.JCR_OAK);
+    private MockSlingHttpServletRequest request;
+    private HarvardNewsSearchServiceImpl harvardNewsSearchService;
+
+    @BeforeEach
+    private void init() {
+        aemContext.registerInjectActivateService(new OakMockSlingRepository());
+        harvardNewsSearchService = aemContext.registerInjectActivateService(new HarvardNewsSearchServiceImpl());
+
+        request = aemContext.request();
+
+        Page page = aemContext.create().page("/content/aemdemo/en");
+
+        Page pageChild1 = aemContext.create().page("/content/aemdemo/en/child1", "/conf/aemdemo/settings/wcm/templates/harvard-single-news-page");
```

Comment:
Please be informed that most often the test content is created through `aemContext.load().json("path-to-json-is-resource-folder", "</content or some other mount point in virtual JCR>"). 
This gives more flexibility and precision. Though your solution also works for the case.

Reply:
thanks, fixed